Sources/core/objects/OnMapBase.cpp

```diff
@@ -13,6 +13,22 @@ IdPtr<ITurf> IOnMapBase::GetTurf()
     return 0;
 }
 
+IdPtr<CubeTile> IOnMapBase::GetTile()
+{
+    IdPtr<IOnMapBase> item = GetId();
+    if (item.IsValid())
```

Comment:
This two lines are unnecessary.
1. Inside object methods you can consider that object itself is valid, so it is not needed to check it.
2. Instead `item->SomeMethod()` you can just write `SomeMethod()`.